packages/components/src/public/QueryModel/utils.ts

```diff
@@ -126,3 +126,24 @@ export function getQueryModelExportParams(
     };
     return getExportParams(type, schemaQuery, exportOptions, advancedOptions);
 }
+
+export function getSelectRowCountColumnsStr(
+    rawColumns?: string | string[],
+    filterArray?: Filter.IFilter[],
+    pkCols?: QueryColumn[]
+): string | string[] {
+    if (!rawColumns || rawColumns === '*') return rawColumns;
+
+    if (filterArray?.length > 0) {
+        const qFilter = filterArray.some(filter => filter.getColumnName() === '*');
+        if (qFilter) return rawColumns;
+    }
+
+    if (pkCols?.length > 0)
+        return pkCols[0].fieldKey;
```

Comment:
this should work fine, but I don't think there would be a harm in including all the pkCols in this case instead of just the first one.